### PR TITLE
Clarify system lifecycle when provided to HttpApp

### PR DESCRIPTION
**Pull request by 2m**
We had received some questions from customers regarding whether the system will be terminated when provided to the HttpApp. This adds a clarification and also points to Coordinated Shutdown for more elaborate shutdown scenarios.

### CODE REVIEW

File: docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md [hunk 1]
```diff
@@ -66,6 +66,8 @@ Scala
 Java
 :   @@snip [HttpAppExampleTest.java](../../../../../test/java/docs/http/javadsl/server/HttpAppExampleTest.java) { #minimal-imports #ownActorSystem }
 
+Here you are responsible for creating and terminating `ActorSystem`. For more fine-grained control over the shutdown of various parts of the application, take a look at @extref[Coordinated Shutdown](akka25-docs:scala/actors.html#coordinated-shutdown) extension which is available since Akka 2.5.0.
```
Comment (raboof) [inline docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md]: Might be nice to be specific and say "When you provide your own `ActorSystem` you are responsible for terminating it." in the first sentence?

File: akka-http-tests/src/test/scala/akka/http/scaladsl/Main.scala [hunk 2]
```diff
@@ -1,5 +0,0 @@
-package akka.http.scaladsl
-
-class Main {
-
-}
```
Comment (ktoso) [inline akka-http-tests/src/test/scala/akka/http/scaladsl/Main.scala]: oh, good catch

File: build.sbt [hunk 3]
```diff
@@ -167,6 +167,7 @@ lazy val docs = project("docs")
         case _                  => "cross CrossVersion.full"
       }),
       "extref.akka-docs.base_url" -> s"http://doc.akka.io/docs/akka/${Dependencies.akkaVersion.value}/%s",
+      "extref.akka25-docs.base_url" -> s"http://doc.akka.io/docs/akka/2.5/%s",
```
Comment (ktoso) [inline build.sbt]: good

File: docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md (hunk 1 above)
Comment (ktoso) [reply docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md]: +1, good to say when exactly instead the "here" :)

Comment (jonas) [reply docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md]: This page has been merged so I think we also need to link to the Java version of the coordinated shutdown page.

Comment (jonas) [inline docs/src/main/paradox/scala/http/routing-dsl/HttpApp.md]: This page has been merged so I think we also need to link to the Java version of the coordinated shutdown page.

**Comment by wsargent** [comment]
Also please add an example showing `postHttpBinding` and how to get an actor system reference from inside `postHttpBinding` (there is no mention of `systemReference` in the docs and it can be null prior to the startServer call)

File: akka-http-tests/src/test/scala/akka/http/scaladsl/Main.scala (hunk 2 above)
Comment (jrudolph) [reply akka-http-tests/src/test/scala/akka/http/scaladsl/Main.scala]: Hehe, yeah crazy long life for that one.

**Comment by jrudolph** [comment]
I think we should also add a note, that `HttpApp` is not the canonical way of starting an akka-http server. It is a tool to get up and running fast. For integrating into already existing Akka applications that already bring their `ActorSystem` using `Http.bindAndHandle` (which is also just a single line of code) is recommended over using `HttpApp`.

**Comment by ktoso** [comment]
That's well worded, +1

**Comment by 2m** [comment]
Thanks for the feedback. Rebased from the latest master and addressed comments.

**Pull request merged**